sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/SimplifyConflictBinaryComparisonSuite.scala

```diff
@@ -0,0 +1,299 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.optimizer
+
+import org.apache.spark.sql.catalyst.dsl.expressions._
+import org.apache.spark.sql.catalyst.dsl.plans._
+import org.apache.spark.sql.catalyst.expressions._
+import org.apache.spark.sql.catalyst.expressions.Literal.FalseLiteral
+import org.apache.spark.sql.catalyst.plans.PlanTest
+import org.apache.spark.sql.catalyst.plans.logical._
+import org.apache.spark.sql.catalyst.rules._
+
+class SimplifyConflictBinaryComparisonSuite extends PlanTest with PredicateHelper  {
+
+  object Optimize extends RuleExecutor[LogicalPlan] {
+    val batches =
+        Batch("Constant Folding", FixedPoint(50),
+          NullPropagation,
+          ConstantFolding,
+          BooleanSimplification,
+          SimplifyBinaryComparison,
+          SimplifyConflictBinaryComparison,
+          PruneFilters) :: Nil
+  }
+
+  val nonNullRelation: LocalRelation = LocalRelation($"a".int.withNullability(false))
+
+  test("EqualTo-related conjunction combinations") {
+    //  a = 1 and a = 2  =====> FalseLiteral
+    val plan1 = nonNullRelation
+      .where(And(EqualTo($"a", Literal.apply(1)), EqualTo($"a", Literal.apply(2))))
+    val actual1 = Optimize.execute(plan1.analyze)
+    val correctAnswer1 = Optimize.execute(nonNullRelation.where(FalseLiteral).analyze)
+    comparePlans(actual1, correctAnswer1)
+
+    // a = 1 and a > 0  ====> a = 1
+    val plan2 = nonNullRelation
+      .where(And(EqualTo($"a", Literal.apply(1)), GreaterThan($"a", Literal.apply(0))))
```

Comment:
Shall we have test case where the literal is of different data type from the attribute?
For example, column `a` is an integer, and there is a predicate `a` = 1 AND a > 0.5

Reply:
updated